src/snapred/backend/recipe/algorithm/DiffractionSpectrumWeightCalculator.py

```diff
@@ -28,76 +28,46 @@ def PyInit(self):
             MatrixWorkspaceProperty("WeightWorkspace", "", Direction.Output, PropertyMode.Mandatory),
             doc="The output workspace to be created by the algorithm",
         )
-        self.declareProperty("DetectorPeakIngredients", defaultValue="", direction=Direction.Input)
         self.declareProperty("DetectorPeaks", defaultValue="", direction=Direction.Input)
 
         self.setRethrows(True)
         self.mantidSnapper = MantidSnapper(self, __name__)
 
-    def chopIngredients(self):
-        pass
-
-    def unbagGroceries(self):
-        self.inputWorkspaceName = self.getPropertyValue("InputWorkspace")
-        self.weightWorkspaceName = self.getPropertyValue("WeightWorkspace")
```

Comment:
Note: chopIngredients was not deleted, it was implemented to perform the task of filling in `self,groupIDs` and `self.predictedPeaks`.  `unbagGroceries` is still the same, but also includes creating the output weight workspace.